primitives/staking/src/lib.rs

```diff
@@ -57,9 +62,10 @@ impl<AccountId, Balance> OnStakerSlash<AccountId, Balance> for () {
 
 /// A struct that reflects stake that an account has in the staking system. Provides a set of
 /// methods to operate on it's properties. Aimed at making `StakingInterface` more concise.
-pub struct Stake<T: StakingInterface + ?Sized> {
+#[derive(Default, Clone, Debug, Eq, PartialEq)]
+pub struct Stake<AccountId: Clone, Balance: Clone> {
```

Comment:
No need to drive `Clone` here, can be a `where` statement only in places where it is logically needed.

Not my opinion, this is a rust idiom, which helps you not repeat where clauses in more places than absolutely needed.

Reply:
How can I call `clone` on `Stake` if it's not derived for it? Or do you actually mean `AccountId` and `Balance` bounds?

Reply:
The latter.